mne_sandbox/connectivity/cfc.py

```diff
@@ -0,0 +1,564 @@
+    f_amp : array, dtype float, shape (2,)
+        The frequency range to use for high-frequency amplitude modulation.
+    ixs : array-like, shape (n_pairs x 2)
+        The indices for low/high frequency channels. PAC will be estimated
+        between n_pairs of channels. Indices correspond to rows of `data`.
+    pac_func : string, list of strings
+        The function for estimating PAC. Corresponds to functions in pacpy.pac.
+        Must be one of ['plv', 'glm', 'mi_canolty', 'mi_tort', 'ozkurt'].
+    ev : array, shape (n_events, 3) | array, shape (n_events,) | None
+        MNE events array. To be supplied if data is 2D and output should be
+        split by events. In this case, tmin and tmax must be provided. If
+        ndim == 1, it is assumed to be event indices, and all events will be
+        grouped together.
+    tmin : float | list of floats, shape (n_pac_windows,) | None
+        If ev is not provided, it is the start time to use in inst. If ev
+        is provided, it is the time (in seconds) to include before each
+        event index. If a list of floats is given, then PAC is calculated
+        for each pair of tmin and tmax.
+    tmax : float | list of floats, shape (n_pac_windows,) | None
+        If ev is not provided, it is the stop time to use in inst. If ev
+        is provided, it is the time (in seconds) to include after each
+        event index. If a list of floats is given, then PAC is calculated
+        for each pair of tmin and tmax.
+    scale_amp_func : None | function
+        If not None, will be called on each amplitude signal in order to scale
+        the values. Function must accept an N-D input and will operate on the
+        last dimension. E.g., skl.preprocessing.scale
+    return_data : bool
+        If True, return the phase and amplitude data along with the PAC values.
+    concat_epochs : bool
+        If True, epochs will be concatenated before calculating PAC values. If
+        epochs are relatively short, this is a good idea in order to improve
+        stability of the PAC metric.
+    n_jobs : int
+        Number of CPUs to use in the computation.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    pac_out : array, list of arrays, dtype float, shape (n_epochs, n_pairs,
+        n_times) The computed phase-amplitude coupling between
+        each pair of data sources given in ixs. If multiple pac metrics are
+        specified, there will be one array per metric in the output list.
+
+    References
+    ----------
+    [1] This function uses the PacPy modulte developed by the Voytek lab.
+        https://github.com/voytekresearch/pacpy
+    """
+    from mne.io.base import _BaseRaw
+    if not isinstance(inst, _BaseRaw):
+        raise ValueError('Must supply either Epochs or Raw')
+    sfreq = inst.info['sfreq']
+    data = inst[:, :][0]
```

Comment:
Soon to appear in our dev docs (once the build passes):

https://circle-artifacts.com/gh/mne-tools/mne-python/999/artifacts/0/home/ubuntu/mne-python/doc/_build/html/generated/mne.io.Raw.html#mne.io.Raw.__getitem__

So `raw[:][0]` is right for raw, and `epochs.get_data()` is right for epochs. But keep in mind this creates a copy, so if this doesn't operate inplace, it might actually be better to access the private `raw._data` object. It's okay to do it here for efficiency since it's private.